src/Collection/Collection.model.ts

```diff
@@ -5,23 +5,48 @@ import { CollectionAttributes } from './Collection.types'
 export class Collection extends Model<CollectionAttributes> {
   static tableName = 'collections'
 
+  static findAll() {
+    return this.query<CollectionAttributes & { item_count: number }>(SQL`
+      SELECT *, (SELECT COUNT(*) FROM ${raw(
+        Item.tableName
+      )} WHERE items.collection_id = collections.id) as item_count
+        FROM ${raw(this.tableName)}
+      `)
+  }
+
+  static findByAllByAddress(address: string) {
+    return this.query<CollectionAttributes & { item_count: number }>(SQL`
+      SELECT *, (SELECT COUNT(*) FROM ${raw(
+        Item.tableName
+      )} WHERE items.collection_id = collections.id) as item_count
+        FROM ${raw(this.tableName)}
+        WHERE eth_address = ${address}
+      `)
+  }
+
   static findByIds(ids: string[]) {
-    return this.query<CollectionAttributes>(SQL`
-    SELECT *
+    return this.query<CollectionAttributes & { item_count: number }>(SQL`
```

Comment:
IMO this should be a new type, specially because the CollectionAttributes containing the amount of items the collection has could be used elsewhere in the code. The problem is naming it and that issue goes again to the same problem, we need better typing in the builder server.
As an example, we need:
- A type for the endpoint's received body (not the case): GetCollectionByAddressRequestBody
- A type for the endpoint's response: GetCollectionByAddressResponseBody
- A type for the DB's entities: CollectionAttributes
- A type for a data object that represents a collection, `Collection` for example. This type and data object could have `item_count` as optional, or we could have another type `CollectionWithItemCount` (not the best name, but it wouldn't ensure that the collection is an object represented as a collection with an item count).

For this case, I would say we should create `Collection` and add `items_count` as optional.

Reply:
So my first attempt was to add it to `CollectionAttributes` but the thing is that it's not really an attribute of the collection type, it's more like an aggregate that comes from the query itself. I'm not a 100% sold on it tho

Reply:
Updated to `CollectionWithItemCount` since `Collection` conflicts with the class declared as the model.

Reply:
umm sorry to chip in this late to the discussion, I have a question, why so many types to represent a collection @LautaroPetaccio? Why should the request and response types be different (I know in the frontend we convert snake_case to camelCase, but that's something that doesn't concern the server). Also I get why the DB attributes and the canonical collection are not the same, but why not just have only the DB attributes and the main collection types ( only 2, instead of 4).

Regarding extending existing types, I think i'm with nico, if the `item_count` property is not something we want to keep in the main type, in my experience the proliferation of `TypeWithSomething` ends up bad, I would rather keep the schemas as canonical as possible. Having it as an optional property could work tho.

Reply:
As of today, we're sending the entire item to the backend, that includes a lot of properties that should not be part of the information needed to update an item, for example:
- blockchain_item_id
- eth_address
- thumbnail
- price
- beneficiary
- total_supply
- is_published
- is_approved
- type
- content_hash
- local_content_hash

If we don't send those properties to the server anymore, we'll end up with a request body that would have its own type (`UpsertItemRequestBody` for example), a type that is different from the response of this or other endpoints that respond with a fully bridged item. This issue shows that we're reusing types that represent two different things.

Although we are not using the properties mentioned before in the server to do an update, the programmer must be aware that they come in the request, as we have only a single schema and type for the input, opening the door to possible overwrites of the database if the input is not sanitized correctly (overwriting et_address for example).

Saying this, there's a difference from what we receive as an input in the various endpoints we have, the things we store and retrieve from the databases and what the endpoints return.